src/EnumSet.php

```diff
@@ -246,6 +246,106 @@ public function isSuperset(EnumSet $other)
     }
 
     /**
+     * Produce a new set with enumerators from both this and other (this | other)
+     * @param EnumSet ...$other Other EnumSet(s) of the same enumeration to produce the union
+     * @return EnumSet
+     */
+    public function union(EnumSet $other)
```

Comment:
let's bump minimum required PHP version to 5.6 for this new feature and make use of variadic functions:
EnumSet::union(EnumSet ...$others)

oldschool PHP < 5.6 users can use the older version of this lib as is